### ISSUE TITLE
-Wpointer-bool-conversion in drivers/soc/tegra/common.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#278]
```
drivers/soc/tegra/common.c:27:16: error: address of array 'match->compatible' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
        while (match->compatible) {
        ~~~~~  ~~~~~~~^~~~~~~~~~
1 error generated.
```

Yay infinite loops! CI actually caught this too since this infinite loop causes our arm kernel not to boot :)

Patch sent: https://lore.kernel.org/lkml/20181122075244.3909-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Patch was accepted: https://git.kernel.org/pub/scm/linux/kernel/git/tegra/linux.git/commit/?id=f24c197ea823059ef36ba15747eb43256fa333b3

But the offending patch has been rebased out and replaced with a better version: https://git.kernel.org/pub/scm/linux/kernel/git/tegra/linux.git/commit/?id=9eb40fa2cd2d1f6829e7b49bb22692f754b9cfe0

I will close this when it shows up in -next (should be next-20181123).

**Comment by nathanchance** [comment]
Fixed in `next-20181123`: https://travis-ci.com/ClangBuiltLinux/continuous-integration/jobs/160438676
